src/packages/core/encoding.cc

```diff
@@ -87,15 +87,15 @@ void f_query_encoding() {
     }
   }
 
-  push_malloced_string(string_copy(res, "f_query_encoding"));
+  push_string(res);
 }
 #endif
 
 #ifdef F_STRING_ENCODE
 void f_string_encode() {
-  auto encoding = sp->u.string;
-  auto data = (sp - 1)->u.string;
-  auto len = SVALUE_STRLEN(sp - 1);
+  auto encoding = sp->u.string->c_str();
```

Comment:
what is the advantage of changing sp->u.string to a std::string?  It is an known good design to make string in the VM immutable, JAVA, python, C# basically all does this, we should consider changing it to to a const char * instead.

Reply:
`std::string` doesn't have any buffer overflows (as long as the implementation within libc++ [whichever is it's real name] is bug-free).
With C-strings one has always to check if there is still enough space available for the intended changes...

And btw.:
I didn't change `u.string` to a `std::string` but a `shared_string` which is a `shared_storage<std::string>` and as such returns the underlying `std::string` only as const reference...